config/ekf_template.yaml

```diff
@@ -0,0 +1,104 @@
+# Hz
+frequency: 30
+sensor_timeout: 0.1
+print_diagnostics: true
+publish_tf: true
+
+# odom frame off
+map_frame: none
+odom_frame: map
+base_link_frame: base_link
+world_frame: map
+
+# odom frame on
+# map_frame: map
+# odom_frame: odom
+# base_link_frame: base_link
+# world_frame: map
+
+# IMU accel and gyro
+# imu0: imu/imu_only
+
+# linearized GPS and IMU fused orientation
+pose0: gps/pose
+
+# magnetometer yaw estimate
+# pose1: imu/mag_pose
+
+# which state variables to update with each sensor measurement
+# [x, y, z,
+#  roll, pitch, yaw,
+#  vx, vy, vz,
+#  vroll, vpitch, vyaw,
+#  ax, ay, az]
+
+# turn off accel x so cmd vel is used for prediction
+# imu0_config: [false, false, false,
+#               false, false, false,
+#               false, false, false,
+#               true, true, true,
+#               false, true, true]
+
+pose0_config: [true, true, true,
+               true, true, true,
+               false, false, false,
+               false, false, false,
+               false, false, false]
+
+# pose1_config: [false, false, false,
+#                false, false, true,
+#                false, false, false,
+#                false, false, false,
+#                false, false, false]
+
+# imu0_remove_gravitational_acceleration: true
+
+# [ADVANCED] If your data is subject to outliers, use these threshold settings, expressed as Mahalanobis distances, to
+# control how far away from the current vehicle state a sensor measurement is permitted to be. Each defaults to
+# numeric_limits<double>::max() if unspecified. It is strongly recommended that these parameters be removed if not
+# required. Data is specified at the level of pose and twist variables, rather than for each variable in isolation.
+# For messages that have both pose and twist data, the parameter specifies to which part of the message we are applying
+# the thresholds.
+# odom0_pose_rejection_threshold: 5
+# odom0_twist_rejection_threshold: 1
+
+use_control: true
+
+# Which velocities are being controlled. Order is vx, vy, vz, vroll, vpitch, vyaw.
+control_config: [true, false, false, false, false, true]
+acceleration_limits: [1.3, 0.0, 0.0, 0.0, 0.0, 3.4]
+deceleration_limits: [1.3, 0.0, 0.0, 0.0, 0.0, 4.5]
+
+# x, y, z, roll, pitch, yaw, vx, vy, vz, vroll, vpitch, vyaw, ax, ay, az
+process_noise_covariance: [0.05, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
```

Comment:
can we measure empirically? Diagonal covariance matrices in EKF do tend to converge in practice but I guarantee you some of these quantities are correlated especially with diff drive kinematics at play.

Reply:
e.g. yaw + any spatial axis

Reply:
I'm not really sure how to measure this empirically especially without any source of ground truth, and I think these values are usually determined just by manually tuning until we get optimal behavior. I'll read some literature and see if I can find an example process noise cov matrix for a differential drive robot to see if its common to correlate some of these things like you say.

Reply:
You have a source of ground truth. The simulator. You can do it on the real robot too, just drive it over a short known trajectory.

Reply:
btw, this comment is not something I completely care about for approval, but I'm saying to add this to the TODOs/issues.

Reply:
ok, will definitely look into it in the future. I don't think the simulator is a very useful source of ground truth for this, since I think its designed to perfectly simulate idealized differential drive dynamics, and therefore it won't tell us much about how noisy the rover's dynamics are, right?

Reply:
You're right about magnitude but I actually think the relative proportions of different entries of the cov mat would be similar

Reply:
just mark this down in an issue and we can resolve it.